.github/workflows/cicd.yaml

```diff
@@ -53,6 +53,39 @@ jobs:
         with:
           path: dist/*.whl
 
+  # Run tests against pydantic v1 until we drop support.
+  test-pydantic-v1:
+    name: test py${{ matrix.python-version }} (pydantic v1) on ${{ matrix.os }}
+    timeout-minutes: 10
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+        python-version: ["3.8", "3.9", "3.10", "3.11"]
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: checkout
+        uses: actions/checkout@v3
+
+      - name: Install poetry
+        run: pipx install poetry
+
+      - name: setup python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: "poetry"
+
+      - name: Install dependencies
+        run: |
+          poetry install
+          poetry run pip install "pydantic<2"
```

Comment:
I suppose something like
```
echo "\n[mypy-pydantic.*]\nignore_missing_imports = True\n" >> mypy.ini
```

would go here [to address @flaviuvadan's idea](https://github.com/argoproj-labs/hera/pull/795#issuecomment-1786175455)?